src/nixos-anywhere.sh

```diff
@@ -316,6 +316,7 @@ is_kexec=\$(if test -f /etc/is_kexec; then echo "y"; else echo "n"; fi)
 is_nixos=\$is_nixos
 is_installer=\$(if [[ "\$is_nixos" == "y" ]] && grep -q VARIANT_ID=installer /etc/os-release; then echo "y"; else echo "n"; fi)
 is_container=\$(if [[ "\$(has systemd-detect-virt)" == "y" ]]; then systemd-detect-virt --container; else echo "none"; fi)
+has_ipv4=\$(if [[ "\$(has ip)" == "n" ]] || ip r g 1.1.1.1 >/dev/null 2>/dev/null; then echo "y"; else echo "n"; fi)
```

Comment:
I would also detect if ipv6 is routable.

```suggestion
has_ipv6_only=\$(if [[ "\$(has ip)" == "n" ]] || ip r g 1.1.1.1 && ip r g 2606:4700:4700::1111 >/dev/null 2>/dev/null; then echo "y"; else echo "n"; fi)
```